### PR TITLE
Support custom secrets

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto-generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 1]
```diff
@@ -180,6 +180,14 @@ capitalization of keys and values).
 If the dict for a section doesn't have a key `Name`, the key for the section will be used as the plugin name for the section.
 This allows avoiding repetition, when it's unnecessary, while still supporting having multiple outputs using the same plugin.
 
+== `secrets`
+
+[horizontal]
+type:: dict
+default:: {}
+
+Create a custom secret.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:189]: Probably makes sense to explain how this creates a custom secret. Also, generally, parameter `secrets` is used to create multiple secrets based on the entries in the dict. I would consider using a different key here (maybe `secret_contents`?)

Comment (DebakelOrakel) [reply docs/modules/ROOT/pages/references/parameters.adoc:189]: True

File: class/defaults.yml [hunk 2]
```diff
@@ -66,6 +66,10 @@ parameters:
           K8S-Logging.Parser: "On"
           K8S-Logging.Exclude: "On"
 
+    # Add additional secrets
```
Comment (simu) [inline class/defaults.yml:69]:
```suggestion
    # Configure a custom secret to deploy
```

File: component/main.jsonnet [hunk 3]
```diff
@@ -162,8 +162,24 @@ local configmap = kube.ConfigMap(params.configMapName) {
...
+      'app.kubernetes.io/name': params.secretName,
+      'app.kubernetes.io/instance': instanceName,
+      'app.kubernetes.io/component': 'fluent-bit',
+      'app.kubernetes.io/managed-by': 'commodore',
+    },
+  },
+  stringData: {
+    [s]: params.secret[s]
+    for s in std.objectFields(params.secret)
+  },
```
Comment (simu) [inline component/main.jsonnet:177]: Hide `data` so Kubernetes is happy:

```suggestion
  },
  data:: {}
```

**Review by simu — APPROVED** [review]

**Pull request merged**